### ISSUE TITLE
Handle when service alias doesn't match the name of the API

### ISSUE THREAD

**Issue by jaypipes** [aws/aws-controllers-k8s#199]
Attempted to build the AWS Step Functions  (SFN) service controller this morning.

The first thing I ran into was the fact that there is no API mode for a service with an alias of `sfn`:

```
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ ./scripts/build-controller.sh sfn
Building Kubernetes API objects for sfn
Error: open ~/.cache/aws-controllers-k8s/src/aws-sdk-go/models/apis/sfn: no such file or directory
Usage:
  ack-generate apis <service> [flags]

Flags:
  -h, --help             help for apis
  -o, --output string    path to directory for service controller to create generated files. Defaults to /$service
      --version string   the resource API Version to use when generating API infrastructure and type definitions (default "v1alpha1")

Global Flags:
      --cache-dir string               Path to directory to store cached files (including clone'd aws-sdk-go repo) (default "/home/jaypipes/.cache/ack-generate")
      --dry-run                        If true, outputs all files to stdout
      --generator-config-path string   Path to file containing instructions for code generation to use
      --services-dir string            Path to directory to output service-specific code (default "/home/jaypipes/go/src/github.com/aws/aws-controllers-k8s/services")
      --templates-dir string           Path to directory with templates to use in code generation (default "/home/jaypipes/go/src/github.com/aws/aws-controllers-k8s/templates")

open ~/.cache/aws-controllers-k8s/src/aws-sdk-go/models/apis/sfn: no such file or directory
```

After some digging, turns out that the AWS Step Functions API models are actually aliased as the ["states" API in the aws-sdk-go/models/apis directory](https://github.com/aws/aws-sdk-go/blob/master/models/apis/states/2016-11-23/api-2.json).

So, I generated the "states" controller:

```
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ ./scripts/build-controller.sh states
Building Kubernetes API objects for states
Generating deepcopy code for states
Generating custom resource definitions for states
Building service controller for states
Generating RBAC manifests for states
activity/descriptor.go:26:2: no matching versions for query "latest"
activity/manager_factory.go:24:2: no matching versions for query "latest"
Error: not all generators ran successfully
run `controller-gen rbac:roleName=ack-controller-role paths=./... output:rbac:artifacts:config=/home/jaypipes/go/src/github.com/aws/aws-controllers-k8s/scripts/../services/states/config//rbac -w` to see all available markers, or `controller-gen rbac:roleName=ack-controller-role paths=./... output:rbac:artifacts:config=/home/jaypipes/go/src/github.com/aws/aws-controllers-k8s/scripts/../services/states/config//rbac -h` for usage
```

Quite predictably, there are failures. If I open up the generated files being complained about above, the `import` code shows the errors clearly:

```
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ head -n 30 services/states/pkg/resource/activity/descriptor.go 
// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License"). You may
// not use this file except in compliance with the License. A copy of the
// License is located at
//
//     http://aws.amazon.com/apache2.0/
//
// or in the "license" file accompanying this file. This file is distributed
// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
// express or implied. See the License for the specific language governing
// permissions and limitations under the License.

// Code generated by ack-generate. DO NOT EDIT.

package activity

import (
	acktypes "github.com/aws/aws-controllers-k8s/pkg/types"
	"github.com/google/go-cmp/cmp"
	"github.com/google/go-cmp/cmp/cmpopts"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	k8sapirt "k8s.io/apimachinery/pkg/runtime"
	k8sctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"

	svcapitypes "github.com/aws/aws-controllers-k8s/services/sfn/apis/v1alpha1"
)

const (
	finalizerString = "finalizers.sfn.services.k8s.aws/Activity"
```

The import paths all refer to the "sfn" service alias that has been discovered by reading the api-2.json models file for the "states" API, but the directory created by `ack-generate` is `services/states/` not `services/sfn`.

The solution to this is two-fold:

1) Change the logic in `ack-generate` and `pkg/model.Helper` to fall back on looking for an API by its Service ID in the api-2.json model file when a directory in `aws-sdk-go/models/apis/` with the supplied alias doesn't exist.

2) Modify `ack-generate` to replace any supplied service alias on the command line with the discovered Service ID after finding the API model file.

**Comment by ChaturvediSulabh** [comment]
Upon running `./scripts/build-controller.sh` states, getting an error similar to #296. Should I log a separate issue for this one?
Error:
`template: pkg/crd_sdk:129:3: executing "pkg/crd_sdk" at <GoCodeRequiredFieldsMissingFromReadOneInput .CRD "r.ko" 1>: error calling GoCodeRequiredFieldsMissingFromReadOneInput: GENERATION FAILURE! there's a required field ActivityArn in Shape DescribeActivityInput that isn't in either the CR's Spec or Status structs! `

**Comment by A-Hilaly** [comment]
@ChaturvediSulabh the issue you're encountering should be solved in #312 :)
building `states` using #312 branch looks good on my machine
```
GOPROXY=direct make build-controller SERVICE=states
go build -tags codegen -ldflags "-X main.version="v0.0.0" -X main.buildHash=22b1c9302630006e42bea22946a83e30abc9b842 -X main.buildDate=2020-09-16T19:31:53Z" -o bin/ack-generate cmd/ack-generate/main.go
./scripts/build-controller.sh states
Building Kubernetes API objects for states
Generating deepcopy code for states
Generating custom resource definitions for states
Building service controller for states
Generating RBAC manifests for states
activity/descriptor.go:27:2: no matching versions for query "latest"
activity/manager_factory.go:24:2: no matching versions for query "latest"
Error: not all generators ran successfully
run `controller-gen rbac:roleName=ack-controller paths=./... output:rbac:artifacts:config=/Users/hilalymh/source/github.com/aws/aws-controllers-k8s/scripts/../services/states/config//rbac -w` to see all available markers, or `controller-gen rbac:roleName=ack-controller paths=./... output:rbac:artifacts:config=/Users/hilalymh/source/github.com/aws/aws-controllers-k8s/scripts/../services/states/config//rbac -h` for usage
Running gofmt against generated code for states
```

**Comment by zacharya** [comment]
@jaypipes I'd like to take this as a first issue and start contributing if you're still looking for help here.

**Comment by jaypipes** [comment]
Hi @zacharya! Actually, we're merged a fix for this already (PR#329). How about finding me on Kubernetes Slack provider-aws channel and we can chat about areas you can contribute in ACK? :)
